src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -20,16 +20,30 @@ export function collectInvalidWordsInIssues(text: string) {
 	const positions: Position[] = [];
 	for (const word of invalidWords) {
 		if (!(allowedWords.includes(word.toLowerCase()))) {
-			const temp: Position = {
-				fromIndex: text.indexOf(word),
-				toIndex: (text.indexOf(word) + word.length),
-			};
-			positions.push(temp);
-			addIssue("INVALID_WORD", temp, {word});
+			const allPositionsOfInvalidWord = getPositionsOfInvalidWord(word, text);
+			for (const pos of allPositionsOfInvalidWord) {
+				positions.push(pos);
+				addIssue("INVALID_WORD", pos, {word});
+			}
 		}
 	}
 }
 
+export function getPositionsOfInvalidWord(invalidWord: string, text: string): Position[] {
```

Comment:
You could simplify this function by knowing that the `String.prototype.indexOf` method can take a second parameter to start searching at. I created pull request #76 to show what it could look like.

I think exported Functions should always have a JSDoc. If I am missing them somewhere then feel free to point it out to me. It is the easiest to create them right away when creating the function. I regretted having pushed the JSDocs to a later time for a seperate pull request.

Reply:
You are right. I will add it once I get to it